app/controllers/application_controller.rb

```diff
@@ -94,7 +94,10 @@ def find_versioned_links
   end
 
   def set_page
-    @page = params[:page].respond_to?(:to_i) ? [1, params[:page].to_i].max : 1
+    @page = Gemcutter::DEFAULT_PAGE && return unless params.key?(:page)
+    redirect_to url_for(page: Gemcutter::DEFAULT_PAGE) unless valid_page_param?
```

Comment:
Do we use any `flash[:notice]` or `flash[:alert]` tips? That will be great place to also show friendly message why user was redirected to first page.

```ruby
redirect_to url_for(page: Gemcutter::DEFAULT_PAGE), notice: 'redirected to page 1 because of invalid page info text'
```

Reply:
added `"Page number is out of range. Redirected to default page."` to ui requests